src/main/java/org/kiwiproject/dropwizard/util/startup/PortAssigner.java

```diff
@@ -0,0 +1,163 @@
+package org.kiwiproject.dropwizard.util.startup;
+
+import static java.util.Objects.isNull;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotNull;
+import static org.kiwiproject.base.KiwiStrings.format;
+import static org.kiwiproject.collect.KiwiLists.first;
+import static org.kiwiproject.dropwizard.util.server.DropwizardConnectors.requireDefaultServerFactory;
+
+import io.dropwizard.jetty.HttpConnectorFactory;
+import io.dropwizard.jetty.HttpsConnectorFactory;
+import io.dropwizard.server.DefaultServerFactory;
+import io.dropwizard.server.ServerFactory;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.config.TlsContextConfiguration;
+import org.kiwiproject.config.provider.TlsConfigProvider;
+import org.kiwiproject.dropwizard.util.exception.NoAvailablePortException;
+import org.kiwiproject.net.LocalPortChecker;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.IntSupplier;
+import java.util.stream.IntStream;
+
+/**
+ * Finds open ports and sets up the application and admin connectors with those ports. There are a couple of opinionated decisions here:
```

Comment:
Do we imply another "opinionated decision" by assuming there is only _one_ application and _one_ admin port? Or is that not true? What happens if there is a secure _and_ non-secure port for both application and admin?

Reply:
Yes, that is an opinionated decision that we only have one of each.  This does not let more than one or both secure and non-secure.

Reply:
I added another line in the javadoc to indicate this decision